transport/ice/IceSession.cpp

```diff
@@ -192,12 +200,28 @@ void IceSession::probeRemoteCandidates(const IceRole role, uint64_t timestamp)
     processTimeout(timestamp);
 }
 
-void IceSession::addProbeForRemoteCandidate(EndpointInfo& endpoint, const IceCandidate& remoteCandidate)
+IceSession::CandidatePair* IceSession::addProbeForRemoteCandidate(EndpointInfo& endpoint,
+    const IceCandidate& remoteCandidate)
 {
     const auto endpointAddress = endpoint.endpoint->getLocalPort();
-    if (endpointAddress.getFamily() != remoteCandidate.address.getFamily())
+    if (endpointAddress.getFamily() != remoteCandidate.address.getFamily() ||
+        endpoint.endpoint->getTransportType() != remoteCandidate.transportType)
```

Comment:
I noticed a bug here that did not really show in logs. If a new peer reflexive TCP candidate showed up we triggered probing from UDP endpoint towards the new candidate.